### PR TITLE
tools/nxstyle: Add the suffix white list

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Found by https://github.com/apache/incubator-nuttx/pull/6226

## Impact
Minor

## Testing
Need merge with https://github.com/apache/incubator-nuttx-apps/pull/1164

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by hartmannathan — COMMENTED** [review]
nxstyle provides for some lowercase naming exceptions:

```
                  /* The coding standard provides for some exceptions of lower
                   * case characters in pre-processor strings:
                   *
                   *   IPv[4|6]    as an IP version number
                   *   ICMPv6      as an ICMP version number
                   *   IGMPv2      as an IGMP version number
                   *   [0-9]p[0-9] as a decimal point
                   *   d[0-9]      as a divisor
                   *   Hz          for frequencies (including KHz, MHz, etc.)
                   */
```

Does it make sense to add `bps` and `ns` / `us` / `ms` to the exception list?

**Comment by xiaoxiang781216** [comment]
Either way is fine for me, what's your opinion @pkarashchenko ?

**Comment by pkarashchenko** [comment]
If there is an exception for units (`Hz`) then probably we can add exception for other units like (`bps`, `us`, `ms`, `ns`). But that will mean that we will add exception for each new unit we introduce in code.

**Comment by pkarashchenko** [comment]
We already have things like
```
#  define SX127X_CMN_PARAMP_PARAMP_250us  (4 << SX127X_CMN_PARAMP_PARAMP_SHIFT)
```
and
```
#  define SX127X_CMN_PLL_BW_75kHz         (0 << SX127X_CMN_PLL_BW_SHIFT)
```
in the code. And despite that `KHz` and `MHz` seem to be acceptable the `kHz` seems to be not. We can try to legitimize those cases via nxstyle exceptions.

**Comment by Ouss4** [comment]
I think it would look weird if we uppercased all of the standard units and their prefixes.  This will lead to confusions.

**Comment by Ouss4** [comment]
I think it would look weird if we uppercased all of the standard units and their prefixes.  This will lead to confusions.

**Comment by xiaoxiang781216** [comment]
Ok, I add a suffix white list to handle these cases, please review again.

File: tools/nxstyle.c [hunk 1]
```diff
@@ -209,6 +209,14 @@ static const char *g_white_prefix[] =
   NULL
 };
 
+static const char *g_white_suffix[] =
+{
+  "kHz",
+  "Mbps",
+  "us",
```
Comment (pkarashchenko) [inline tools/nxstyle.c:216]: do we need may `"ms"` and `"ns"` to be added as well?

**Comment by Ouss4** [comment]
Can we add more entries to the list?
If I remember correctly we once changed a macro that had `mA` to `MA`, which is funny to say the least. :)

**Comment by xiaoxiang781216** [comment]
> Can we add more entries to the list? If I remember correctly we once changed a macro that had `mA` to `MA`, which is funny to say the least. :)

all these macros end with MA now.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by pkarashchenko** [comment]
> Can we add more entries to the list? If I remember correctly we once changed a macro that had `mA` to `MA`, which is funny to say the least. :)

Now we can add exception and rename back :)

**Pull request merged**